misc/validators/validator-puppet

```diff
@@ -18,6 +18,32 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301,
 # USA.
 #
+#
+# NOTE about puppet version
+# ------------------------------------------------------------------------
+# This driver script doesn't take care the version of puppet.
+# As the result, you may get much warnings on some platform
+# though the results of validations are "valid".
+#
+# As of writing this comment, puppet5 becomes popular.
+# In #1909, @ahakanbaba has fixed much of test cases to be
+# valid as puppet5 files. In other hands, only puppet4
+# is available from official Fedora28 repository.
+# With using puppet4 as puppet, this validator reports
+# much warnings like:
+#
```

Comment:
As of writing this comment, puppet5 is becoming popular. In #1909, @ahakanbaba has fixed much of test cases to be valid puppet5 files. Even tough that PR exists, we are still sticking with puppet4 in our automated tests. One of our test platforms is Fedora28 and its distribution has puppet4 at this time. As the note about warnings above mentions, in validation runs using puppet4 you may see the following Errors prompts that do not fail the validation.